### PR TITLE
fix(tests): fix obvious things in `tests.mk`

### PR DESCRIPTION
**Pull request by sergio-mena**
As per our internal discussion today, here are some changes I think we should have.

Feel free to comment on further changes you may want to have in `tests.mk`

---

#### PR checklist

- ~[ ] Tests written/updated~
- ~[ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~
- ~[ ] Updated relevant documentation (`docs/` or `spec/`) and code comments~
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: tests.mk [hunk 1]
```diff
@@ -47,22 +47,29 @@ test_release:
...
 test:
 	@echo "--> Running go test"
-	@go test -p 1 $(PACKAGES) -tags deadlock,bls12381
+	@go test -p 1 $(PACKAGES) -tags bls12381
 .PHONY: test
 
 test_race:
 	@echo "--> Running go test --race"
-	@go test -p 1 -v -race $(PACKAGES)
+	@go test -p 1 -race $(PACKAGES)
```
Comment (cason) [inline tests.mk:58]: No bls here?

File: tests.mk [hunk 2]
```diff
@@ -47,22 +47,29 @@ test_release:
...
+
+# The format statement filters out all packages that don't have tests.
+# Note we need to check for both in-package tests (.TestGoFiles) and
+# out-of-package tests (.XTestGoFiles).
+$(BUILDDIR)/packages.txt:$(GO_TEST_FILES) $(BUILDDIR)
+	go list -f "{{ if (or .TestGoFiles .XTestGoFiles) }}{{ .ImportPath }}{{ end }}" ./... | sort > $@
+
+split-test-packages:$(BUILDDIR)/packages.txt
+	split -d -n l/$(NUM_SPLIT) $< $<.
+test-group-%:split-test-packages
```
Comment (cason) [inline tests.mk:74]:
```suggestion

# Used, in special, by the GitHub CI, in order to run tests in parallel 
test-group-%:split-test-packages
```

**Review by cason — APPROVED** [review]
+1

File: tests.mk (hunk 1 above)
Comment (sergio-mena) [reply tests.mk:58]: Good catch

File: tests.mk [hunk 3]
```diff
@@ -47,22 +47,29 @@ test_release:
...
-	@go test -p 1 -v -race $(PACKAGES)
+	@go test -p 1 -race $(PACKAGES)
 .PHONY: test_race
 
 test_deadlock:
-	@echo "--> Running go test --deadlock"
-	@go test -p 1 -v  $(PACKAGES) -tags deadlock
-.PHONY: test_race
+	@echo "--> Running go test with deadlock support"
+	@go test -p 1 -v  $(PACKAGES) -tags deadlock,bls12381
```
Comment (sergio-mena) [inline tests.mk:63]:
```suggestion
	@go test -p 1 $(PACKAGES) -tags deadlock,bls12381
```

**Pull request merged**